Add a bottom app bar to the floating action button motion demo.

DaveShuckerow opened the pull request:
@HansMuller Here's the demo.

Commit:
Add support to move the fab between positions.

Commit:
Motion demo for the FAB works between center and end floating.

Commit:
Add a Material curve to the offset animation.

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Move the fab position into an object

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Updates to docs

Commit:
Updates to docs

Commit:
Merge branch 'fab-motion' of github.com:DaveShuckerow/flutter into fab-motion

Commit:
Fix a lint on the bottom sheet type

Commit:
Merge branch 'fab-motion' of github.com:DaveShuckerow/flutter into fab-motion

Commit:
Add a ScaffoldGeometry class

Commit:
Improve the documentation

Commit:
Improve the documentation

Commit:
Add a fab motion animator

Commit:
Add position and scale animations

Commit:
FAB entrance and motion animations work

Commit:
Get started on FAB motion

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Make fab animation work properly.

Commit:
Change the fab animator to be stored in the state of the scaffold.

Commit:
Add a layout test

Commit:
Fix spacing being off

Commit:
Fix the entrance/exit animation test.

Commit:
Add a textDirection to the layout delegate.

Commit:
Fix const constructor lint checks

Commit:
Add toStrings for the fab positioner/animator

Commit:
Add a toString for CurveTween

Commit:
Merge branch 'tostring' into fab-motion

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Rename the ScaffoldGeometry for FAB layout to ScaffoldPrelayoutGeometry.

Commit:
Change the fab motion demo icon to a simple add icon.

Commit:
Add tests and a custom fab positioner to the demo.

Commit:
Do not start the fab's motion animation when the fab is null.

Commit:
Adjust the code to pass the new tests.

Commit:
Rename for in response to Hans' comment.

Commit:
Revert the tabs fab demo

Commit:
Use timeDilation, and clean up the animation code a little.

Commit:
Clean up the prelayout geometry docs and ctr order

Commit:
Cleanup fab transition widget code

Commit:
Clean up comments on Scaffold, add cross-references between the two geometries

Commit:
Explain the fab motion animation scheduling better

Commit:
Add a const to the fab motion demo

Commit:
Make the fab animation never jank by keeping track of where to move the fab to in the future.

Commit:
Add a default fab positioner constant

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Add space after comma in the demo

Commit:
Add boilerplate dartdoc to all const constructors

Commit:
Comment improvement

Commit:
Rename 'fabSize' to 'floatingActionButtonSize'

Commit:
Rename 'fabSize' to 'floatingActionButtonSize'

Commit:
Rename 'fabSize' to 'floatingActionButtonSize'

Commit:
Clean up the prelayout geometry object's dartdoc

Commit:
Clean up the prelayout geometry object's dartdoc

Commit:
Remove extraneous comment

Commit:
Change possessive uses of Scaffold's to use dartdoc-compatible [Scaffold]'s

Commit:
Rename the horizontalFabPadding to an expansion

Commit:
Clean up controller cleanup and setState usage

Commit:
Animate instead of lerp

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Make the fab position animation use offsets instead of animations

Commit:
Merge branch 'fab-motion' of github.com:DaveShuckerow/flutter into fab-motion

Commit:
Streamline the fab motion demo

Commit:
Set up the animator to start from a reasonable place when interrupting animations.

Commit:
Doc cleanup on the new animation interruption

Commit:
Expand some uses of fab and clean up constants

Commit:
Expand remaining public uses of fab to floating action button

Commit:
Expand remaining public uses of fab to floating action button

Commit:
Expand on the documentation for the fab positioner and animator

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Refactor animations to broadcast the position properly.

Commit:
Add the ability to turn on and off the fab to the motion demo.

Commit:
Remove unused code

Commit:
Change the fab animator to animate even when the fab is exitting.

Commit:
Remove extra positioner.

Commit:
Merge branch 'master' of https://github.com/flutter/flutter into fab-motion

Commit:
Apps -> Applications in docs

Commit:
Merge branch 'fab-motion' of github.com:DaveShuckerow/flutter into fab-motion

Commit:
Explain the scale animation.

Commit:
Name the child parameter in the animated builder

Commit:
RTL before LTR

Commit:
Wrap the AppBar in the example code

Commit:
const the fab motion demo name

Commit:
Start a test against animation jumps

Commit:
Test for jumps in the fab motion animation

Commit:
Dont initialize values to null

Commit:
Use constants, fix spacing from some of Hans' comments

Commit:
Clarify the relationship between fab positioners and prelayout geometries

Commit:
Explain the fab animmator a bit better

Commit:
Explain the animation progress in the fab animation

Commit:
Explain the animation restart better

Commit:
Explain the animation restart better

Commit:
Explain the prelayout geometry better

Commit:
Explain that height is a vertical distance.

Commit:
Explain the horizontal fab padding

Commit:
Update the scaffold size description to explain what happens when a wild keyboard appears

Commit:
Remove print statements

Commit:
Update the scaffold geometry with information about it being available at paint time.

Commit:
In one step of a transition

Commit:
Explain how the top-start fab positioner works

Commit:
Explain how the top-start fab positioner works

Commit:
Refactor the scaffold layout to just pass a padding instead of a bottom, top, start and end.

Commit:
Refactor the scaffold layout to just pass a padding instead of a bottom, top, start and end.

Commit:
Action buttons with with custom positioners.

Commit:
Add a rotation animation example.

Commit:
Use a swap animation to show swapping between two different animations.

Commit:
Use a swap animation to show swapping between two different animations.

Commit:
Add an example for the size animations.

Commit:
2018 copyright

Commit:
Extra empty line

Commit:
Return new Scaffold

Commit:
Merge branch 'fab-motion' of github.com:DaveShuckerow/flutter into fab-motion

Commit:
Extra blank line fix

Commit:
All its contents have been laid out

Commit:
Position the fab

Commit:
Explain what the scaffold geometry is for.

Commit:
Move asserts to different lines

Commit:
The scaffoldsize will not

Commit:
Initial rename of FabPositioners to FloatingActionButtonLocation

Commit:
Rename comments in example to refer to location instead of positioner.

Commit:
Rename fabpositioner to location in tests and in the scaffold field

Commit:
Finish removing references to positioner in scaffold code.

Commit:
Split the fab location and animation out into a separate file.

Commit:
Make things more private

Commit:
Import foundation instead of meta

Commit:
Const curve instead of final.

Commit:
Extraneous space

Commit:
Add docked positions to the motion demo.

Commit:
Add a Bottom App Bar to the fab motion demo.

Commit:
Add bab to the name of the fab motion demo

Commit:
Add bab to the name of the fab motion demo

Commit:
Merge branch 'master' into bab-demo

xster committed:
Revert "Add a simple convenience test task runner for VSCode (#15417)" (#15705)

This reverts commit c8fd99fb3313c73eb03210a3af037cd2d150ef51.

gspencergoog commented on examples/flutter_gallery/lib/demo/material/fab_motion_demo.dart:
```diff
@@ -145,3 +156,71 @@ class _TopStartFloatingActionButtonLocation extends FloatingActionButtonLocation
+
+/// Provider of common logic for [FloatingActionButtonLocation]s that
+/// dock to the [BottomAppBar].
+abstract class _DockedFloatingActionButtonLocation extends FloatingActionButtonLocation {
```
Could this be a mixin instead?  Not that it needs to be, but since it's just providing functionality and no state, it seems like a candidate for that.

gspencergoog commented on examples/flutter_gallery/lib/demo/material/fab_motion_demo.dart:
```diff
@@ -145,3 +156,71 @@ class _TopStartFloatingActionButtonLocation extends FloatingActionButtonLocation
   }
 }
+
+/// Provider of common logic for [FloatingActionButtonLocation]s that
```
These can be normal comments, since this is a private class.

gspencergoog committed:
Add a slider demo, and a text theme for SliderThemeData (#15620)

This adds a slider demo with a custom theme to the gallery.

In the process of adding this, I decided to add a text theme to the SliderThemeData, since it's a pain to change the text style on the value indicator otherwise.

xqwzts committed:
Allow null flutter schema in pubspec (#15690)

Commit:
Added Indonesian (id) and Norwegian (no) locales (#15708)

Hixie committed:
Move dialogs when keyboard pops up (#15426)

Fixes #7032

Hixie committed:
Update some READMEs under dev/. (#15712)

Hixie committed:
Document the sample API doc analyzer (#14740)

Hixie committed:
Improve documentation regarding padding for dialogs. (#15399)

xster committed:
Add a way to generically go back on both Material and Cupertino scaffold pages (#15602)

* Add a way to generically go back on both Material and Cupertino scaffold pages

* review

* lint

* review

* Fix text assert text

xster committed:
Remove the 'demo' wording from gallery (#15640)

jonahwilliams committed:
Provide an initial rotation of the ExpansionIcon if isExpanded = true on initState (#15657)

* rotate expansion icon fully if initial state expanded = true

* add test case for initially expanded

* remove unused import

* Update expand_icon_test.dart

add linebreak

* import dart:math as math

xster committed:
Add debug disable flags for various performance affecting layers (#15548)

vilkh3m committed:
add Samsung SM-J730F to isLocalEmulator (#15517)

Add another rel device to isLocalEmulator to make working flutter run --release

goderbauer committed:
Support password fields for a11y (#15497)

* Support password fields for a11y

* rename to obscured

* Roll engine to c3ab0c9143029f0267a05b99effbfbd280a4901b

a14n committed:
apply upcomming lint avoid_renaming_method_parameters (#15526)

gspencergoog committed:
Fix package preparation on Windows (#15720)

a-siva committed:
Roll engine to version 6280adbfb1f9f63cdc6179b9b78634add1e4f2e3 (#15729)

* roll engine version to 6280adbfb1f9f63cdc6179b9b78634add1e4f2e3

* Correctly handle previewDart : false setting in AnalysisServer and
AnalyzeOnce classes.

mraleph committed:
Report an error if compilation during testing times out. (#15745)

* Report an error if compilation times out instead of waiting forever.

* Remove braces

argumentofperiapsis committed:
Fixed flutter create's "package" template's error in pubspec.yaml. (#14977)

kevmoo committed:
Update LICENSE file so it can be detected (#15353)

See https://help.github.com/articles/licensing-a-repository/#detecting-a-license

jacob314 committed:
Make the inspector handle widgets with non-invertible transforms gracefully. (#15761)

Commit:
Fixed a typo in translations_test.dart: Norwegian LC is "no" (#15746)

mravn-google committed:
Add missing awaits (#15595)

jonahwilliams committed:
Traverse dependencies and dev dependencies separately (#15581)

* traverse dependencies and dev dependencies separately

* separate dev and reg deps

* update pubspecs

* add checksum string and update transitive dep string.  Add additional verify command to check checksum

* cleanup comments

* add missing space, quote, run verify-onlu from bot script

* rerun tool with fixed string

* ensure correct working directory and bump versions

* flutter not flutterRoot

jonahwilliams committed:
Pass shadowColor param to SceneBuilder (#15714)

* pass shadowColor param to SceneBuilder

* add doc comment and diagnostic node for shadowColor

* Add missing shadowColor param

a14n committed:
add new lints of linter-0.1.44 and enable unnecessary_parenthesis (#15585)

jason-simmons committed:
Remove an error that is caught at compile time in Dart 2 (#15797)

aam committed:
Pass --no-preview-dart-2 for dart1 tests. (#15800)

xster committed:
Update gallery drawer version (#15774)

Commit:
InputDecorator should clip fill color to its border (#15795)

thekeenant committed:
Fix call on null when interpolating gradient stops (#15656)

Also adds test for interpolating gradient stops.

xster committed:
Add more docs and examples for implementers of RenderSliver (#15419)

* Add more docs and examples for implementers of RenderSliver

* add more typical scenarios to describe scrollOffset

* review

* tweak a word

xster committed:
Cupertino pull to refresh part 1: sliver and a simple indicator widget builder (#15324)

* Gallery scaffolding

* Started RenderSliver

* demo and initial hookup

* Cleaned up demo more and scaffolding basic sliver->widget communication structure.

* works

* states and default indicator building works

* start adding docs

* added an alignment setting optimized the sliver relayout mechanism

* tested a default bottom aligned sized indicator

* Added a bunch of tests

* more fixes and more tests

* Finished the tests

* Add docs

* Add more doc diffing wrt material pull to refresh

* Mention nav bar synergy

* add more asserts

* review 1

* Fix mockito 2 / dart 2 / strong typed tests

* review

* Remove the vscode config

* review

Hixie committed:
Fix RTL icons (#15095)

gspencergoog committed:
Updating example for MultiChildLayoutDelegate (#15597)

xster committed:
Remove cupertino pull to refresh from transitions test (#15813)

gspencergoog committed:
Creating chips custom renderer, updating visual look. (#15596)

This updates the visual look of the Chip class, so that it scales properly in the face of text scale (and label widget size) changes, and bases its height off of the label widget's height, constraining the other widgets it contains to be the same height.

To do this properly, I had to implement a custom render object that will measure the height of the label, so the guts of this class are basically rewritten.

In addition, to allow the circle avatar to increase in size when the chip does, I added minRadius and maxRadius arguments to it, and I updated its color handling to use the light/dark primary colors of the theme in a smart way instead of just using black and white.

Updated and added tests.

a14n committed:
enable lint avoid_relative_lib_imports (#15780)

mravn-google committed:
Stop Gradle daemon before attempting to delete Flutter project (#15819)

mravn-google committed:
Fix path to .bat file (#15824)

Commit:
Remove package:http from Flutter (#15416)

* use HttpOverrides and dart:io HttpClient in flutter

* add missing package:http dependency

* update flutter packages and remove comment about createHttpClient from flutter_test

* move byte loading logic to common class, move string parsing logic to base class

* addAll doesn't work for a Uint8List

* use bytes.setRange

* undo addition to hello_world

* add newline to end of binding.dart

* and a newline for hello world

* refactor to function and add tests

* address comments on unknown length case

* alignment

* sort alaphabetically

* rename convertResponse to consolidateClientHttpClientResponseBytes.  Add header

* fix alignment in test

gspencergoog committed:
Changing the JSON metadata format for the website. (#15834)

Commit:
Extended ButtonTheme, RoundedRectangleBorder, gallery buttons demo (#15723)

Hixie committed:
Clean up the existing Navigator API. (#15718)

This is not a grand refactor yet, it's just cleaning up what we have
already, so that people who keep using this API (e.g. dialogs) have
something coherent to deal with.

The major changes are that Navigator and NavigatorState have the same
API now, that most of the examples use `<void>` instead of `<Null>`,
that the navigator observer can see replaces, and that the `settings`
is moved from ModalRoute to Route. I also cleaned up some of the API
documentation.

Hixie committed:
Add Form.onChanged, and rename FormState.onChanged (#15405)

awdavies committed:
[fuchsia_remote_debug_protocol] Adds the fuchsia remote debug protocol package.

Adds a package that will support connecting to and debugging devices with multiple instances of the Dart VM, and operate on multiple Isolates and Flutter Views.

Currently supports connecting to all existing Dart VMs and extracting information about all Flutter Views running across each VM.

mravn-google committed:
Change to absolute path (#15827)

gspencergoog committed:
Updating checkmark animation (#15850)

xster committed:
Do not push -pre versions to stores (#15845)

xster committed:
Re-enable usage of TransformLayers with text in RenderListWheelViewport (#15625)

* Re-enable usage of TransformLayers with text in RenderListWheelViewport

* add test

* linter fix

Commit:
Extended Floating Action Button (#15841)

xster committed:
Add more doc linking the material and cupertino refresh controls (#15842)

* Add more doc linking the material and cupertino refresh controls

* review

yjbanov committed:
CONTRIBUTING.md: updates for Dart 2; move parts to flutter/engine (#15874)

* CONTRIBUTING.md: updates for Dart 2; move parts to flutter/engine

* address comments

Commit:
Marked flutter_gallery_instrumentation_test flaky (#15891)

srawlins committed:
Remove tear-off mocking in refresh_test (#15869)

* Remove tear-off mocking in refresh_test

* Comments

* Analyzer fix

* Final final final final mushroom mushroom!

* Fix for Dart 1 AND Dart 2 semantics; anonymous closure over tear-off

* Final final

aam committed:
Send relative uri when target physical file path is specified. (#15887)

Commit:
Update AndroidManifest to include screenLayout|density in config changes (#15871)

* android templates should add density and screenLayout to config changes

* update the rest of the manifests

gspencergoog committed:
Post submit review changes for chip (#15835)

gspencergoog committed:
Address post-submit comments for sliders (#15836)

* Address post-submit comments

* Added dispose protection for the animation controllers.

sbaranov committed:
Removed obsolete step in for rolling a new engine

zanderso committed:
[fuchsia] Fix package/flutter BUILD.gn deps (#15951)

a-siva committed:
Roll engine to version dd6f46c485192f4506035088c9065b9f5dbba9ab (#15885)

Roll engine to version dd6f46c485192f4506035088c9065b9f5dbba9ab

aam committed:
Move mockito to 3.0.0-alpha. (#15949)

* Move mockito to 3.0.0-alpha.

This is needed so it matches mockito version used internally.

* Pick up +2 for the fixes

* Fix whenReturn in fuchsia_remote_debug_protocol

* more thenReturn becomes thenAnswer

* more thenReturn becomes thenAnswer

* more thenReturn becomes thenAnswer

aam committed:
Revert "Move mockito to 3.0.0-alpha. (#15949)" (#15979)

This reverts commit a7e6225820719eec08ae3c79ce73ebeb8ec33523 as there are failing tests.

cbracken committed:
Revert "Roll engine to version dd6f46c485192f4506035088c9065b9f5dbba9ab (#15885)" (#15980)

This caused a test failure in the integration_ui_ios devicelab test.

```
stdout: [   +3 ms] 00:02 [32m+0[0m: end-to-end test Ensure keyboard dismissal resizes the view to original size[0m
stdout: [+8129 ms] 00:11 [32m+0[0m[31m -1[0m: end-to-end test Ensure keyboard dismissal resizes the view to original size [1m[31m[E][0m[0m
stdout: [   +5 ms]   DriverError: Failed to fulfill Tap: Flutter application not responding
stdout: [        ]   Original error: TimeoutException after 0:00:07.500000: Future not completed
stdout: [        ]   Original stack trace:
stdout: [        ]   #0      FlutterDriver._sendCommand (package:flutter_driver/src/driver/driver.dart:324:18)
stdout: [        ]
stdout: [        ]   #1      FlutterDriver.tap (package:flutter_driver/src/driver/driver.dart:368:11)
stdout: [        ]
stdout: [        ]   #2      main.. (file:///[... snip ...]/flutter/dev/integration_tests/ui/test_driver/keyboard_resize_test.dart:34:20)
```

This reverts commit 067be92ae8fe2e82529c4b23db8df58d05028558.

Hixie committed:
Update gallery about box. (#15954)

cbracken committed:
Roll engine to 2114a88b5ee0c5240f2aec6cde164798d00b61b4 (#15987)

Includes:
* Provide asset lookup key on ios (flutter/engine#4817)
* Load kernel files from the script URI in the isolate create callback (flutter/engine#4828)
* Remove some unneeded profiling flags (flutter/engine#4827)
* Remove the canvas save/restore calls in Paragraph::Paint (flutter/engine#4822)

cbracken committed:
Roll engine to e102ae3d28eace7f350ece188187c2566920e654 (#15988)

Includes:
* Roll Skia (flutter/engine#4841)

Commit:
Corrected default tab indicator color (#15959)

Commit:
Make gallery tests more robust (#15957)

cbracken committed:
Roll engine to 01be4c6017f98532eea2456ea422e2c285e219c8 (#15990)

Includes:
* Fix GPUSurfaceGL includes on Linux (flutter/engine#4849)
* Slightly improve the docs for the internal _futurize method (flutter/engine#4847)
* Fix supported color type check on iOS simulators (flutter/engine#4846)
* Fix GPUSurfaceGL includes on macOS (flutter/engine#4844)
* Handle Apple-specific GLES headers (flutter/engine#4843)
* [fuchsia] Fix up includes of Zircon headers (flutter/engine#4842)
* VulkanSurface and GPUSurfaceGL no longer use GrPixelConfig (flutter/engine#4814)

cbracken committed:
Roll engine to 3e877d371a359097d7af28aa6e85026fa2318baa (#15992)

Includes:
* Remove declaration for undefined GPUSurfaceGL::SelectPixelConfig (flutter/engine#4857)
* Vulkan swap chain setup no longer uses GrPixelConfig and doesn't incl…
* Revert dart roll 20180322 (flutter/engine#4855)
* Roll back pub upgrade. (flutter/engine#4854)
* Roll flutter 20180322a (flutter/engine#4852)
* Document Window.defaultRouteName a little better. (flutter/engine#4851)
* Roll dart to 95e9e890a9 (flutter/engine#4850)

cbracken committed:
Apply Mockito-related type fixes (#16002)

Type-related fixes in flutter_driver and fuchsia_remote_debug_protocol
tests.

Commit:
Fixed an uninitialized bool in the gallery FAB demo (#16012)

sbaranov committed:
Support for decimal and signed numeric keyboard (#15846)

* Support for decimal and signed numeric keyboard

* Comments

* Rebase.

* Roll engine to dd6f46c485192f4506035088c9065b9f5dbba9ab

Commit:
Updated memory_nav_test with FlutterDriver.scrollUntilVisible() (#16009)

tvolkert committed:
Make AppContext immutable and race-free (#15984)

This updates AppContext per the recommendations in #15352

Fixes #15352

tvolkert committed:
Mark test as non-flaky now that it's been fixed (#16033)

aam committed:
Again move mockito to 300plus2 (#15981)

* Revert "Revert "Move mockito to 3.0.0-alpha. (#15949)" (#15979)"

This reverts commit e59651f925917ed44a9bf27c37e80fc9992c0890.

* More thenReturn to thenAnswer when mocking Futures

* Revert "More thenReturn to thenAnswer when mocking Futures"

This reverts commit 194d2cf41754ad171efd6bb5898cf73bce5a75ab as we are reverting engine roll.

* Revert "Revert "More thenReturn to thenAnswer when mocking Futures""

This reverts commit 52c9e96b304b220f66d78a252fccea0c8e0ef24c.

* Add dependency override

* Fix issue

tvolkert committed:
Make AppContext API not be prone to Future oddities (#16034)

Follow-up comments to #15984

mravn-google committed:
Experiment to understand test flakiness (#15872)

xster committed:
Let travis test archiving per commit too (#16004)

cbracken committed:
Roll engine to 6473f1b106485cb0b4ea569af383173daeef8895 (#16032)

Includes:
* Use newer Skia API for constructing mask blur filters (flutter/engine#4875)
* Retry fixing vulkan include (flutter/engine#4874)

xster committed:
Revert "Let travis test archiving per commit too" (#16050)

* Revert "Roll engine to 6473f1b106485cb0b4ea569af383173daeef8895 (#16032)"

This reverts commit 844ced010a19e7375e246f6d3f001bdc0d7305fd.

* Revert "Let travis test archiving per commit too (#16004)"

This reverts commit df5905f3823acaf5596bad909d27a66b3b5f9c70.

xster committed:
Let iOS auto-signing tool handle Googler scenario (#16010)

* Let iOS auto-signing tool handle Googler scenario

* Forgot to update tests

xster committed:
Revert "Re-enable usage of TransformLayers with text in RenderListWheelViewport (#15625)" (#16042)

This reverts commit 1f5d904182f866102fe40a3e640863f9888783e7.

tvolkert committed:
Remove writelnStderr (#16058)

Paying off technical debt...

Commit:
improve Flutter build commands (#15788)

add --buildNumber and --buildName to flutter build like
flutter build apk --buildNumber=42 --buildName=1.0.42

Commit:
Support for disabling TextField, TextFormField (#16027)

tvolkert committed:
Move Cache.flutterRoot initialization up sooner (#16057)

It's required to be set before we detect local engine.

Was broken by #15984

najeira committed:
AppBar shows CloseButton even on custom PageRoute (#15643)

* AppBar accepts custom PageRoute

* remove unused import

* update AUTHORS

Commit:
Address comments from #5714 (#15886)

* add simple test case for passing shadow color through physical layer

* Update physical_model_test.dart

add newline

goderbauer committed:
Fix FAB Tooltip semantics label (#16072)

mravn-google committed:
Remove flaky flag from non-flaky test (#16075)

jason-simmons committed:
URL encode asset paths to support reserved characters in paths (#16046)

Fixes https://github.com/flutter/flutter/issues/14864

Commit:
Make sure `flutter update-packages --verify-only` has a non-zero exit code (#15962)

* make sure --verify-only logs error messages and fix typo

* ensure non zero exit code

* undo random pubspec change

tvolkert committed:
Revert "Make sure `flutter update-packages --verify-only` has a non-zero exit code (#15962)" (#16089)

This reverts commit cfda7a65487dbad44bbeeca1516f84b7166bded6.

cbracken committed:
Roll engine to 9af82c5a1a79ae02477db13e721c8365f31a4840 (#16088)

Includes:
* libtxt: move to the next run if the current run ends before the start of the line block (flutter/engine#4891)
* Rolled Dart SDK to e3051175192c83fafa3636c0f16d804734bb285a (flutter/engine#4877)

aam committed:
Roll engine. (#16087)

Changes since last roll:
```
ec1e96864 Revert pub roll from recent dart roll. (#4904)
ac552fba4 Port flutter content handler to FIDL2. (#4903)
94cdd8103 Roll dart. (#4902)
af63732de Support `Path.addCircle` (#4783)
564c6a5a9 Add hint to use ninja with goma (#4894)
```
```

jcollins-g committed:
Update dartdoc to v0.18.1 and fix API readme (#16080)

* Update dartdoc to v0.18.1 and fix API readme, borrowing from docs.flutter.io

Hixie commented:
@DaveShuckerow are you looking for a review? if you don't want to commit this, should we close it? let us know what you want from this PR :-)

aam committed:
Roll engine to c903c217a1a8206cdebdab1703b52ec6180edf37. (#16116)

This is to pick up updated dart roll.

xster committed:
Let travis test archiving per commit too (#16004) (#16107)

xster committed:
Fix gallery build ruby script (#16121)

* Fix gallery build ruby script

* Also print a message when not deploying

Commit:
Update logic for parsing sdk version number (#15918)

* add logic to parse 11.3 sim runtime major version

* add null aware and bump group number

* add comment describing version

DaveShuckerow commented:
I will close this for now.  Hans and I are discussing the next steps offline.

Commit:
Update indeterminate LinearProgressIndicator (#16106)

xqwzts committed:
Better error message when TextField is not in a Material (#16147)

Added a call to debugCheckHasMaterial() at the start of TextField's build method.

xster committed:
Let CupertinoTabScaffold's tab be programmatically selectable (#16040)

* Let CupertinoTabScaffold's tab be programmatically selectable

* Re-use the tab bar's index instead

* review

Commit:
Merge branch 'bab-demo' of github.com:DaveShuckerow/flutter into bab-demo

Commit:
Merge the fab motion demo and Amirs bottom app bar demo.

Commit:
Alphabetize the demos

Commit:
Simplify the menu by changing radio buttons to dropdowns

Commit:
Remove the fab motion demo to finish moving it to the bab demo.

Commit:
Remove the fab motion demo to finish moving it to the bab demo.

Commit:
Comment clarifications>
'.

Commit:
Move some code around

Commit:
Comment cleanup

DaveShuckerow commented:
@HansMuller I'm reopening to check in the bottom app bar demo.  I made a few changes to Amir's code to make the page easier to use.  

It properly integrates between the bottom app bar and the floating action button locations.
